dp_mobility_report/model/preprocessing.py

```diff
@@ -44,12 +44,13 @@ def _validate_columns(df: pd.DataFrame) -> pd.DataFrame:
         raise ValueError("Column 'lng' must be present in data.")
     if not pd.core.dtypes.common.is_float_dtype(df[const.LNG]):
         raise TypeError("Column 'lng' is not of type float.")
-    if const.DATETIME not in df.columns:
+    if const.DATETIME not in df.columns and timestamps:
         raise ValueError("Column 'datetime' must be present in data.")
-    try:
-        df.loc[:, const.DATETIME] = pd.to_datetime(df[const.DATETIME])
-    except Exception as ex:
-        raise TypeError("Column 'datetime' cannot be cast to datetime.") from ex
+    if const.DATETIME not in df.columns and timestamps:
```

Comment:
we don't need the exact same "if" clause twice

Reply:
That is right. Modify this condition to if timestamps.